### PR TITLE
HPlus: added low battery information

### PR DESCRIPTION
**Pull request by albertsal**
I added the notification of low battery HPlus ... specifically, I used
the HPlusConstants.DATA_STATS event on hplusSupport

I also included the status of the device in the list of
gbdevice -> getDeviceInfos

### CODE REVIEW

**Comment by albertsal** [comment]
hi @jpbarraca  i have correct pr

**Comment by jpbarraca** [comment]
I don't feel confortable with such specific fields being added to GBDevice as I think a more generic approach could be followed. As an example, through a collection of generic slots that each device handler could use to provide device specific information.

@cpfeiffer : can you comment on this?

**Comment by albertsal** [comment]
I saw that it lacks a global summary of activities so I thought I'd add it.
This usually has to be on the first screen just have not found a valid space

**Comment by jpbarraca** [comment]
@albertsal I actually can agree with the concept but I think the implementation can be different.

**Comment by albertsal** [comment]
ok I accept suggestions

**Comment by jpbarraca** [comment]
Different devices will have different metrics to be presented. 
Adding specific properties to GBDevice such as Calories, Steps, etc... is only valid for HPlus devices and GBDevice is a more generic class.

Therefore, IMHO your PR makes sense if the Device panel shows data from a collection of items with a known format (interface). This interface could have a title, value (and a level?). The level (which I think is clearly optional) could be used to display some values in a different color if they represent a critical or warn situation.

Other approach would be for GBDevice to be abstract and each device implements its interface in a device specific manner.

However this requires a discussion about the GB user interface, which is already in progress.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/impl/GBDevice.java [hunk 1]
```diff
@@ -78,6 +78,15 @@ public GBDevice createFromParcel(Parcel source) {
     private String mBusyTask;
     private List<ItemWithDetails> mDeviceInfos;
 
+    private int mStep = 0;
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/impl/GBDevice.java]: @jpbarraca is right -- this is device specific and should not be there. However, there is already support for adding "any kind" of additional information to GBDevice. See method GBDevice#addDeviceInfo(). Can you try to use that?

Comment (albertsal) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/impl/GBDevice.java]: ok i try

**Comment by albertsal** [comment]
hi @jpbarraca
I changed all the logic now in use #addDeviceInfo as suggested by
@cpfeiffer
I think now it's ok

**Comment by cpfeiffer** [comment]
Yes, this looks much better, thanks! From my point of view only the method name should be improved (Extra_info() => processExtraInfo() or something like that.

Thanks again :+1:

**Comment by cpfeiffer** [comment]
BTW, you don't *have* to revert commits. You can also just add your changes in another commit and we will squash all of them together before merging.

**Comment by albertsal** [comment]
you're right, but I thought it was more properly the revert, since I have changed all

**Comment by cpfeiffer** [comment]
So from my point of view, we can merge it. @jpbarraca, what do you think? Can we merge this or do we need a better mechanism for displaying the information?

**Comment by jpbarraca** [comment]
I think it is ok.

**Comment by cpfeiffer** [comment]
Uh. I was just about to merge it. Now you added another commit and we need to take another round of review. How about resetting the branch to the previous commit, force-pushing it and opening another branch for the lastChange thing?

**Comment by albertsal** [comment]
I'm sorry I thought you were in time to add this piece ... I take it out and add it later

**Pull request merged**